hw/top_earlgrey/data/ip/chip_rv_timer_testplan.hjson

```diff
@@ -0,0 +1,59 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{
+  name: chip_rv_timer
+  testpoints: [
+    {
+      name: chip_sw_timer
+      desc: '''Verify the timeout interrupt assertion.
+
+            - Configure RV_TIMER to generate an interrupt after a set timeout.
+            - Issue a WFI to wait for the interrupt to trigger.
+            - Service the interrupt when it triggers; verify that it came from rv_timer.
+            - Verify that the interrupt triggered only after the timeout elapsed.
+            '''
+      stage: V2
+      si_stage: SV2
+      features: ["RV_TIMER.ENABLE", "RV_TIMER.CONFIG", "RV_TIMER.COMPARE", "RV_TIMER.INTERRUPT",
+                 "RV_TIMER.COUNTER"]
+      tests: ["chip_sw_rv_timer_irq"]
+    }
+    {
+      name: tick_configuration
+      desc: '''Verify that the timer can be configured to generate a system tick.
+
+            - Configure the timer to generate a tick of `T` microseconds long.
+            - Enable the timer.
+            - Start a busy loop of 1 second based on the `mcycleh` CSR.
+            - Read the number of ticks `n`.
+            - Verify that `n * T ~= 1 second` within 5% of tolerance.
+            - Repeat for T in:
+                [1, 10, 100 and 1000]
+            '''
+      stage: V3
+      si_stage: SV2
+      features: ["RV_TIMER.ENABLE", "RV_TIMER.CONFIG", "RV_TIMER.COMPARE"]
+      tests: []
+    }
+    {
+      name: riscv_isa_csrs
+      desc: '''Verify that the timer counter and compare registers can be read and written via
+            the registers `mtime` and `mtimecmp` and that `mtime` will wrap to zero when the limit
+            is reached.
+
+            - Enable interrupts.
+            - Set the compare register to zero by writing to `mtimecmp` CSR.
+            - Set the counter to `UINT64_MAX - 500 milliseconds` by writing to `mtime` CSR.
```

Comment:
This is a tremendously large value! The tests need to run in a reasonably short amount of time.

Reply:
Maybe the description is not clear, but my intention is to set the counter to a value close to wrap value. So it would take about 500 ms to run. Does that make sense?

Reply:
Ah, thanks. Clear now.